content/languages/javascript/authoring.md

```diff
@@ -103,7 +103,8 @@ describe("Generated test cases", () => {
 });
 ```
 
-This technique is liked by authors familiar with testing frameworks that provide parameterized or generated test cases out of the box, like NUnit or JUnit. However, some caution is needed when this approach is used. Test suites organized like this can become large and can flood the test output panel with many entries, making it unreadable or causing performance problems in client browsers.
+This technique is liked by authors familiar with testing frameworks that provide parameterized or generated test cases out of the box, like NUnit or JUnit. However, some caution is needed when this approach is used. Test suites organized like this can become large and can flood the test output panel with many entries, making it unreadable or causing performance problems in client browsers. Additionally, to avoid problems with captured variables, declare them with `const` or `let`, and never use `var` or global variables.
```

Comment:
It's fine as long as the value doesn't "leak" and get overwritten. So `forEach` (like the example above) is fine as well. Immutable globals shouldn't cause issues either.

Reply:
I have no good idea how to word it, mostly because I think I am too noob with JS and I do not fully understand the problem. Could you please suggest something?

Reply:
See https://stackoverflow.com/a/750506 for a list of solutions and https://2ality.com/2015/02/es6-scoping.html scoping details.

To be honest, I'm not sure if we should include this in our docs anymore. JavaScript has gotchas like most languages, and we can't cover all. Modern JS has `let`/`const` and `var` should be avoided anyway.

The Python decorator is definitely worth mentioning because it's really weird.